packages/multi-test/src/wasm.rs

```diff
@@ -1,620 +1,706 @@
-            block,
-            api,
-            storage_factory,
+    fn query(
+        &self,
+        storage: &dyn Storage,
+        querier: &dyn Querier,
+        block: &BlockInfo,
+        request: WasmQuery,
+    ) -> Result<Binary, String> {
+        match request {
+            WasmQuery::Smart { contract_addr, msg } => {
+                let addr = self
+                    .api
+                    .addr_validate(&contract_addr)
+                    .map_err(|e| e.to_string())?;
+                self.query_smart(addr, storage, querier, block, msg.into())
+            }
+            WasmQuery::Raw { contract_addr, key } => {
+                let addr = self
+                    .api
+                    .addr_validate(&contract_addr)
+                    .map_err(|e| e.to_string())?;
+                Ok(self.query_raw(addr, storage, &key))
+            }
+            q => panic!("Unsupported wasm query: {:?}", q),
         }
     }
 
-    pub fn set_block(&mut self, block: BlockInfo) {
-        self.block = block;
-    }
-
-    // this let's use use "next block" steps that add eg. one height and 5 seconds
-    pub fn update_block<F: Fn(&mut BlockInfo)>(&mut self, action: F) {
-        action(&mut self.block);
-    }
-
-    /// Returns a copy of the current block_info
-    pub fn block_info(&self) -> BlockInfo {
-        self.block.clone()
+    fn execute(
+        &self,
+        storage: &mut dyn Storage,
+        router: &Router<C>,
+        block: &BlockInfo,
+        sender: Addr,
+        msg: WasmMsg,
+    ) -> Result<AppResponse, String> {
+        let (resender, res) = self.execute_wasm(storage, router, block, sender, msg)?;
+        self.process_response(router, storage, block, resender, res, false)
     }
 
-    pub fn store_code(&mut self, code: Box<dyn Contract<C>>) -> usize {
+    fn store_code(&mut self, code: Box<dyn Contract<C>>) -> usize {
         let idx = self.codes.len() + 1;
         self.codes.insert(idx, code);
         idx
     }
 
-    pub fn cache(&'_ self) -> WasmCache<'_, C> {
-        WasmCache::new(self)
+    fn sudo(
+        &self,
+        contract_addr: Addr,
+        storage: &mut dyn Storage,
+        router: &Router<C>,
+        block: &BlockInfo,
+        msg: Vec<u8>,
+    ) -> Result<AppResponse, String> {
+        let res = self.call_sudo(contract_addr.clone(), storage, router, block, msg)?;
+        self.process_response(router, storage, block, contract_addr, res, false)
     }
+}
 
-    pub fn query(&self, querier: &dyn Querier, request: WasmQuery) -> Result<Binary, String> {
-        match request {
-            WasmQuery::Smart { contract_addr, msg } => {
-                self.query_smart(Addr::unchecked(contract_addr), querier, msg.into())
-            }
-            WasmQuery::Raw { contract_addr, key } => {
-                self.query_raw(Addr::unchecked(contract_addr), &key)
-            }
-            q => panic!("Unsupported wasm query: {:?}", q),
+impl<C> WasmKeeper<C>
```

Comment:
In passing: Shouldn't `SimpleBank` be callled `BankKeeper` for consistency?

Reply:
Done